src/blob/core.rs

```diff
@@ -541,13 +542,18 @@ impl RawRecords {
         }
     }
 
-    async fn load(mut self) -> Result<Option<Vec<RecordHeader>>> {
+    async fn load(mut self, validate_data: bool) -> Result<Option<Vec<RecordHeader>>> {
```

Comment:
From design point of view, it is better to pass this argument to `start` function and store its value as a field in `RawRecords` struct